### PR TITLE
CMake: Fix linking with compiler other than MSVC.

### PR DESCRIPTION
**Pull request by Biswa96**
Previously, MSVC specific pragma comment are used to link with ws2_32 library.
But those does not work with MinGW gcc and clang. Hence, use CMake's own
target_link_libraries() to link with ws2_32 for Win32 platform conditionally.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 2.8.12)
 # Defer enabling C and CXX languages.
 project(AWSLC NONE)
 
-if(WIN32)
+if(MSVC)
```
Comment (dkostic) [inline CMakeLists.txt:6]: I didn't understand why is this line changed, could you please clarify?

Comment (Biswa96) [reply CMakeLists.txt:6]: Because cl.exe is Visual Studio compiler and this PR is about when not compiling with Visual Studio.

**Comment by andrewhop** [comment]
Thanks for fixing this! We currently only test [Windows with MSVC](https://github.com/awslabs/aws-lc/tree/main/tests/ci#test-locations) and that build is passing with this change. Windows with Clang is currently untested but we're happy to take this fix now and investigate adding more Windows builds in the future.

How did you test this change?

If you don't mind me asking, what is your use case for AWS-LC?

**Comment by Biswa96** [comment]
Is 'just poking around' considered as use case? 😛

**Review by andrewhop — APPROVED** [review]

**Review by dkostic — APPROVED** [review]
